pkg/kubelet/kubelet.go

```diff
@@ -957,7 +959,17 @@ func (kl *Kubelet) Run(updates <-chan kubetypes.PodUpdate) {
 	}
 
 	// Start volume manager
-	go kl.volumeManager.Run(wait.NeverStop)
+	readyCh := make(chan bool, 1)
+	go kl.volumeManager.Run(readyCh, wait.NeverStop)
+	// Ready channel is set when all the sources are ready
+	go func() {
+		for {
+			if kl.sourcesReady.AllReady() {
+				readyCh <- true
+				break
```

Comment:
So let's do this: 
If possible, pass sourcesReady directly and have VolumeManger call AllReady when it needs to.
If that is not possible,  introduce a sleep here (50ms? 100ms?) so that it's not polling in a tight loop and consuming CPU.